### PR TITLE
fix: add check to avoid converting 0 to undefined

### LINKED ISSUE #794: it.each/describe.each break when single-dimensional array contains 0 (numeric zero)
### Describe the bug

Migrating tests from jest to Vitest, the way `describe.each` and `it.each` function is buggy. According to [the jest docs](https://jestjs.io/docs/api#describeeachtablename-fn-timeout), using `describe.each` or `it.each` should internally convert a single-dimensional array to a multidimensional array. For example, `[1, 2, 0]` would be converted to `[[1], [2], [0]]`. This is a very common scenario for simple tests.

Unfortunately, using the single-dimensional array syntax with Vitest v0.4.1, testing with `[1, 2, 0]` fails when running the test for `0` (numeric zero). It seems that the value gets converted to `undefined`!

I did find the comment in https://github.com/vitest-dev/vitest/pull/640 indicating that template literal usage is intentionally left out, but it makes no mention of single-dimensional arrays, which I see as being far more common. If single-dimensional arrays are also being omitted, then I think the docs need to be updated to call out these differences.

### Reproduction

```js
[1, 2, 0].forEach((num) => {
	it(`${num} is a number (foreach)`, () => {
		expect(typeof num).toEqual('number');
	});
});

it.each([1, 2, 0])('%s is a number (it.each 1d)', (num) => {
	expect(typeof num).toEqual('number');
});

it.each([
	[1],
	[2],
	[0],
])('%s is a number (it.each 2d)', (num) => {
	expect(typeof num).toEqual('number');
});

describe.each([1, 2, 0])('%s (describe.each 1d)', (num) => {
	it(`${num} is a number (describe.each 1d)`, () => {
		expect(typeof num).toEqual('number');
	});
});

describe.each([
	[1],
	[2],
	[0],
])('%s (describe.each 2d)', (num) => {
	it(`${num} is a number (describe.each 2d)`, () => {
		expect(typeof num).toEqual('number');
	});
});
```

```bash
     √ 1 is a number (foreach)
     √ 2 is a number (foreach)
     √ 0 is a number (foreach)
     √ 1 is a number (it.each 1d)
     √ 2 is a number (it.each 1d)
     × %s is a number (it.each 1d)
     √ 1 is a number (it.each 2d)
     √ 2 is a number (it.each 2d)
     √ 0 is a number (it.each 2d)
     √ 1 (describe.each 1d) (1)
     √ 2 (describe.each 1d) (1)
     ❯ %s (describe.each 1d) (1)
       × undefined is a number (describe.each 1d)
     √ 1 (describe.each 2d) (1)
     √ 2 (describe.each 2d) (1)
     √ 0 (describe.each 2d) (1)
```

Failing tests result in `AssertionError: expected 'undefined' to deeply equal 'number'`

### System Info

```shell
  System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
    Memory: 25.69 GB / 31.36 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  npmPackages:
    @vitejs/plugin-vue: ^2.2.2 => 2.2.2
    vite: ^2.8.4 => 2.8.4
    vitest: ^0.4.1 => 0.4.1
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by togami2864**
resolve #794

### CODE REVIEW

File: packages/vitest/src/utils/base.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export function clone<T>(val: T): T {
  */
 
 export function toArray<T>(array?: Nullable<Arrayable<T>>): Array<T> {
-  array = array || []
+  if (array === null || array === undefined) array = []
```
Comment (Demivan) [inline packages/vitest/src/utils/base.ts:56]: You can use `array ??= []` to simplify this assignment.

Comment (userquin) [reply packages/vitest/src/utils/base.ts:56]: @Demivan for such simple changes you can modify it directly here editing the file

Comment (Demivan) [reply packages/vitest/src/utils/base.ts:56]:
```suggestion
  array ??= []
```

@userquin Thanks, I'm not reviewing on Github often enough to remember this :)

Comment (userquin) [reply packages/vitest/src/utils/base.ts:56]: ok I'll change it, the PR LGTM

Comment (userquin) [reply packages/vitest/src/utils/base.ts]: upps, what's the typescript version? it is failing on tasks, on local it is working with node 16, it seems node 14

Comment (Demivan) [reply packages/vitest/src/utils/base.ts]: It should be supported from 3.7 or something like that. Are we not compiling those files? It should convert to `array ?? (array = []);` after compilation.

Comment (userquin) [reply packages/vitest/src/utils/base.ts]: no, the tests are using the monorepo not the dist folder: it compiles to `array ?? (array = []);`

Comment (userquin) [reply packages/vitest/src/utils/base.ts]: I cannot revert the commit from here

**Review by Demivan — APPROVED** [review]

**Review by userquin — APPROVED** [review]

**Comment by userquin** [comment]
sorry guys, next time I'll check lint on local

**Pull request merged**